src/rules/no-empty-file/index.js

```diff
@@ -0,0 +1,27 @@
+import {
+  ruleMessages,
+  report,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "no-empty-file"
+
+export const messages = ruleMessages(ruleName, {
+  rejected: "Unexpected empty file",
+})
+
+export default function (actual) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, { actual })
+    if (!validOptions) { return }
+
+    if (root.toString().replace(/\s/g, "").length > 0) { return }
```

Comment:
A regex test might be more straightforward here, right? `/^\s*$`/.test(root.toString())`